lib/prompt_writer.rb

```diff
@@ -34,23 +26,38 @@ def replay
 
   private
 
+  attr_reader :std_out 
+
   def divider
     " | " 
   end
 
-  def display_cell(cells, index)
-    cells[index] == "empty" ? one_based(index).to_s : cells[index].to_s
+  def format_board(board)
+    board_to_display = ""  
+    cell_number = 0 
+    board.grid_for_display.each do |row|
```

Comment:
As said before, there are usually good alternatives to iteratively building data using `.each`. Instead of mutating the `board_to_display` string, you can create an array of lines and `join` them with a newline. For example:
```ruby
result = ""
["apples", "oranges"].each do |fruit|
  result += fruit
  result += "\n"
end
result
```
is equivalent to
```ruby
["apples", "oranges"].join("\n")
```